### ISSUE TITLE
Crash on queryRenderedFeatures with boolean properties

### ISSUE THREAD

**Issue by mkv123** [mapbox/mapbox-gl-native#6093]
**Platform:** Android
**Mapbox SDK version:** Latest snapshot

### Steps to trigger behavior

 1. Define some custom data with complex geojson types (we have dicts and arrays in the geojson properties we've uploaded to mapbox studio)
 2. setStyleUrl to that custom style
 3. move the map so that queryRenderedFeatures should return the data

### Expected behavior
queryRenderedFeatures returns features.

### Actual behavior
Crash. Every time.

```
08-19 20:22:23.242 17790-17790/com.mapbox.mapboxandroiddemo D/mbgl: {pboxandroiddemo}[JNI]: nativeMoveBy
08-19 20:22:23.243 17790-17790/com.mapbox.mapboxandroiddemo D/mbgl: {pboxandroiddemo}[Android]: NativeMapView::notifyMapChange()
08-19 20:22:23.243 17790-17790/com.mapbox.mapboxandroiddemo D/mbgl: {pboxandroiddemo}[JNI]: nativeGetCameraValues
08-19 20:22:23.244 17790-17790/com.mapbox.mapboxandroiddemo D/mbgl: {pboxandroiddemo}[JNI]: nativeQueryRenderedFeatures for Point
08-19 20:22:23.251 17790-17790/com.mapbox.mapboxandroiddemo E/libc++abi: terminating with uncaught exception of type jni::PendingJavaException
08-19 20:22:23.251 17790-17790/com.mapbox.mapboxandroiddemo A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 17790 (pboxandroiddemo)
```

Modified SimpleMapViewActivity to replicate issue:

```java
        mapView.getMapAsync(new OnMapReadyCallback() {
            @Override
            public void onMapReady(final MapboxMap mapboxMap) {
                mapboxMap.setStyleUrl("mapbox://styles/...customstyle...");

                mapboxMap.setCameraPosition(new CameraPosition.Builder().target(new LatLng(60.186132, 24.827881)).zoom(18).build());

                mapboxMap.setOnCameraChangeListener(new MapboxMap.OnCameraChangeListener() {
                    @Override
                    public void onCameraChange(CameraPosition position) {
                        PointF center = new PointF(mapView.getX()+mapView.getWidth()/2, mapView.getY()+mapView.getHeight()/2);
                        System.out.println("Features: "+mapboxMap.queryRenderedFeatures(center));
                    }
                });

**Comment by tobrun** [comment]
cc @ivovandongen

**Comment by mkv123** [comment]
Apparently this isn't caused by the complex features we have, as scrubbing the properties of dicts and lists doesn't fix the issue. So I'm a bit at a loss what is causing the crash, except for having some custom tileset at that location. 

I'll update the topic to be more general.

**Comment by mkv123** [comment]
After quite some detective work, it appears anything except numbers and strings cause the crash.

Here is a very minimal geojson to replicate the crash with a boolean argument:

```json
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {
        "inBuilding": false
    },
      "geometry": {
        "type": "Polygon",
        "coordinates":
          [
          [
            [
              24.834015369415283,
              60.19064675189364
            ],
            [
              24.834015369415283,
              60.19154276294592
            ],
            [
              24.836440086364746,
              60.19154276294592
            ],
            [
              24.836440086364746,
              60.19064675189364
            ],
            [
              24.834015369415283,
              60.19064675189364
            ]
          ]
        ]
      }
    }
  ]
}
```

**Comment by ivovandongen** [comment]
@mkv123 Thanks for digging into this. There was an issue with conversion of booleans specifically indeed. Fix is in #6256
